File: Heat/Controllers/ProductsController.vb
Imports System.Data.Entity
Imports System.Net
Imports DataTables.AspNet.Core
Imports Heat.Models
Imports Heat.Manager


Namespace Controllers
    Public Class ProductsController
        Inherits Controller

        Private _db As IHeatDBContext
        Private _pm As ProductManager

        Public Sub New(dbcontext As IHeatDBContext)
            _db = dbcontext
            _pm = New ProductManager(_db)
        End Sub

        ' GET: Products
        Function Index() As ActionResult
            Return View(_db.Products.ToList())
        End Function

        ' GET: Products/Details/5
        Function Details(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim product As Product = _db.Products.Find(id)
            If IsNothing(product) Then
                Return HttpNotFound()
            End If
            Return View(product)
        End Function

        ' GET: Products/Create
        Function Create() As ActionResult
            Return View()
        End Function

        ' POST: Products/Create
        'To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        'more details see http://go.microsoft.com/fwlink/?LinkId=317598.
        <HttpPost()>
        <ValidateAntiForgeryToken()>
        Function Create(<Bind(Include:="ID,SKU,Description,UnitPrice,Cost,ReorderLevel")> ByVal product As Product) As ActionResult
            If ModelState.IsValid Then
                _db.Products.Add(product)
                _db.SaveChanges()
                Return RedirectToAction("Index")
            End If
            Return View(product)
        End Function

        ' GET: Products/Edit/5
        Function Edit(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim product As Product = _db.Products.Find(id)
            If IsNothing(product) Then
                Return HttpNotFound()
            End If
            Return View(product)
        End Function

        ' POST: Products/Edit/5
        'To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        'more details see http://go.microsoft.com/fwlink/?LinkId=317598.
        <HttpPost()>
        <ValidateAntiForgeryToken()>
        Function Edit(<Bind(Include:="ID,SKU,Description,UnitPrice,Cost,ReorderLevel")> ByVal product As Product) As ActionResult
            If ModelState.IsValid Then
                '_db.Entry(product).State = EntityState.Modified
                _db.SetModified(product)
                _db.SaveChanges()
                Return RedirectToAction("Index")
            End If
            Return View(product)
        End Function

        ' GET: Products/Delete/5
        Function Delete(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim product As Product = _db.Products.Find(id)
            If IsNothing(product) Then
                Return HttpNotFound()
            End If
            Return View(product)
        End Function

        ' POST: Products/Delete/5
        <HttpPost()>
        <ActionName("Delete")>
        <ValidateAntiForgeryToken()>
        Function DeleteConfirmed(ByVal id As Integer) As ActionResult
            Dim product As Product = _db.Products.Find(id)
            _db.Products.Remove(product)
            _db.SaveChanges()
            Return RedirectToAction("Index")
        End Function

        <HttpGet>
        Public Function GetPagedProducts(request As IDataTablesRequest) As ActionResult
            If ModelState.IsValid Then
                Return _pm.GetPagedProducts(request)
            Else
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
        End Function


        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            If (disposing) Then
                _db.Dispose()
            End If
            MyBase.Dispose(disposing)
        End Sub
    End Class
End Namespace

File: Heat.ConvertedToC#/Controllers/ProductsController.cs
using System.Linq;
using System.Web.Mvc;
using System.Net;
using DataTables.AspNet.Core;
using Heat.Models;
using Heat.Manager;


namespace Heat.Controllers
{
    public class ProductsController : Controller
	{

		private IHeatDBContext _db;

		private ProductManager _pm;
		public ProductsController(IHeatDBContext dbcontext)
		{
			_db = dbcontext;
			_pm = new ProductManager(_db);
		}

		// GET: Products
		public ActionResult Index()
		{
			return View(_db.Products.ToList());
		}

		// GET: Products/Details/5
		public ActionResult Details(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			Product product = _db.Products.Find(id);
			if ((product == null)) {
				return HttpNotFound();
			}
			return View(product);
		}

		// GET: Products/Create
		public ActionResult Create()
		{
			return View();
		}

		// POST: Products/Create
		//To protect from overposting attacks, please enable the specific properties you want to bind to, for 
		//more details see http://go.microsoft.com/fwlink/?LinkId=317598.
		[HttpPost()]
		[ValidateAntiForgeryToken()]
		public ActionResult Create(		[Bind(Include = "ID,SKU,Description,UnitPrice,Cost,ReorderLevel")]
Product product)
		{
			if (ModelState.IsValid) {
				_db.Products.Add(product);
				_db.SaveChanges();
				return RedirectToAction("Index");
			}
			return View(product);
		}

		// GET: Products/Edit/5
		public ActionResult Edit(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			Product product = _db.Products.Find(id);
			if ((product == null)) {
				return HttpNotFound();
			}
			return View(product);
		}

		// POST: Products/Edit/5
		//To protect from overposting attacks, please enable the specific properties you want to bind to, for 
		//more details see http://go.microsoft.com/fwlink/?LinkId=317598.
		[HttpPost()]
		[ValidateAntiForgeryToken()]
		public ActionResult Edit(		[Bind(Include = "ID,SKU,Description,UnitPrice,Cost,ReorderLevel")]
Product product)
		{
			if (ModelState.IsValid) {
				//_db.Entry(product).State = EntityState.Modified
				_db.SetModified(product);
				_db.SaveChanges();
				return RedirectToAction("Index");
			}
			return View(product);
		}

		// GET: Products/Delete/5
		public ActionResult Delete(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			Product product = _db.Products.Find(id);
			if ((product == null)) {
				return HttpNotFound();
			}
			return View(product);
		}

		// POST: Products/Delete/5
		[HttpPost()]
		[ActionName("Delete")]
		[ValidateAntiForgeryToken()]
		public ActionResult DeleteConfirmed(int id)
		{
			Product product = _db.Products.Find(id);
			_db.Products.Remove(product);
			_db.SaveChanges();
			return RedirectToAction("Index");
		}

		[HttpGet()]
		public ActionResult GetPagedProducts(IDataTablesRequest request)
		{
			if (ModelState.IsValid) {
				return _pm.GetPagedProducts(request);
			} else {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
		}


		protected override void Dispose(bool disposing)
		{
			if ((disposing)) {
				_db.Dispose();
			}
			base.Dispose(disposing);
		}
	}
}
